src/vs/workbench/services/backup/node/backupFileService.ts

```diff
@@ -14,84 +14,173 @@ import { IEnvironmentService } from 'vs/platform/environment/common/environment'
 import { IFileService } from 'vs/platform/files/common/files';
 import { TPromise } from 'vs/base/common/winjs.base';
 
+export interface IBackupsFileModel {
+	resolve(backupRoot: string): TPromise<IBackupsFileModel>;
+
+	add(resource: Uri, versionId?: number): void;
+	has(resource: Uri, versionId?: number): boolean;
+	remove(resource: Uri): void;
+	clear(): void;
+}
+
+// TODO@daniel this should resolve the backups with their file names once we have the metadata in place
+export class BackupsFileModel implements IBackupsFileModel {
```

Comment:
Maybe `BackupFilesModel` would be clearer to indicate this deals with multiple files?